draft-how-does-mutant-work.markdown

```diff
@@ -0,0 +1,1630 @@
+As part of a [presentation](https://github.com/troessner/talks/blob/master/exploiting_rubys_ast_a_love_story_in_three_chapters/presentation.html) I'm hoping to give at the end of this year about abstract syntax trees and how you can leverage them I started to look into how the [Mutant](https://github.com/mbj/mutant) gem works. `Mutant` is the by far most advanced gem for doing mutation testing in Ruby. I was (and still am) amazed how incredibly well built it is on all levels, ranging from the "big picture" architecture down to the "nuts and bolts" source code.
+
+When it comes to mutation testing I learned one thing. It's complex. Really complex. And as a consequence, `Mutant` is complex. 
```

Comment:
do you mean `complex` or `complicated` ?
http://english.stackexchange.com/questions/10459/what-is-the-difference-between-complicated-and-complex